cli/command/service/logs.go

```diff
@@ -70,28 +75,45 @@ func runLogs(dockerCli *command.DockerCli, opts *logsOptions) error {
 		Tail:       opts.tail,
 	}
 
-	client := dockerCli.Client()
+	// this variable was original called "client" which conflicts with the
+	// imported "github.com/docker/docker/client" package.
+	cli := dockerCli.Client()
 
-	service, _, err := client.ServiceInspectWithRaw(ctx, opts.service)
-	if err != nil {
-		return err
-	}
+	var (
+		padding      = 1
+		responseBody io.ReadCloser
+	)
 
-	responseBody, err := client.ServiceLogs(ctx, opts.service, options)
+	service, _, err := cli.ServiceInspectWithRaw(ctx, opts.target)
 	if err != nil {
-		return err
+		// if it's any error other than service not found, it's Real
+		if !client.IsErrServiceNotFound(err) {
+			return err
+		}
+		_, _, err := cli.TaskInspectWithRaw(ctx, opts.target)
```

Comment:
This doesn't look addressed but it has been below. I moved the logs call to after the error check.